src/script/descriptor.cpp

```diff
@@ -1360,7 +1360,7 @@ class MiniscriptDescriptor final : public DescriptorImpl
         for (const auto& arg : m_pubkey_args) {
             providers.push_back(arg->Clone());
         }
-        return std::make_unique<MiniscriptDescriptor>(std::move(providers), miniscript::MakeNodeRef<uint32_t>(*m_node));
+        return std::make_unique<MiniscriptDescriptor>(std::move(providers), miniscript::MakeNodeRef<uint32_t>(m_node->Clone()));
```

Comment:
The PR description asserts:
> Multipath descriptors requires performing a deep copy

Would be happy if you cared to add an elaboration on why that is.

It seemed to me like it should be safe to just have another `shared_ptr` point to the same `const Node`. Unless something on the outside could own a non-const reference into the node hierarchy and mutate it that way? In that case maybe it would be more robust for the `MiniscriptDescriptor`-ctor to be the one ensuring it holds a unique reference by doing the `node->Clone()` there instead (if `.use_count() > 1`).

Was able to avoid the crash in #30864 using only this on top of the parent commit of this PR:
```suggestion
        return std::make_unique<MiniscriptDescriptor>(std::move(providers), m_node);
```
Probably causes other bugs that are obvious for those who understand more of the context. Change passes both unit and non-extended functional tests though.

Reply:
It's not strictly necessary but I decided to do it this way to follow the pattern used for all other descriptors. The multipath expanded descriptors are treated as separate descriptors everywhere, and doing a deep copy retains that behavior which will allow for future changes that may modify specific descriptors.

Reply:
How come the shallow copy in the version before the PR is causing a problem in this case though?

It seems the default-generated copy-ctor for `Node` used before was somehow ending up with corrupt/leaked data, but I've been unable to spot what it is. Can't see any slicing going on. Is something funky being done to the `shared_ptr`s?

Regardless, it might be worth adding:
```
    Node(const Node&) = delete;
    Node(Node&&) = delete;
    Node& operator=(const Node&) = delete;
    Node& operator=(Node&&) = delete;
```

Reply:
Not entirely sure either.

Reply:
The shallow copy in the version before this PR is causing a problem because we mess with a `Node`'s `subs` in the destructor to avoid a recursion stack overflow:
https://github.com/bitcoin/bitcoin/blob/2c90f8e08c4cf44d4c1ef3dde0e7f7991b8b9390/src/script/miniscript.h#L513-L524

In the scriptpubkeyman fuzz target using the seed from #30864 we would `Parse` the `tr(%17/<2;3>,l:pk(%08))` multipath descriptor into a `parsed_descs` vector of 2 `Descriptor` pointers. We would use the first of those descriptors (`tr(xpub..../2,l:pk(4bdf....))#jmgpm0u8`) and move it into a `WalletDescriptor`. When exiting the `CreateWalletDescriptor` function the `parsed_descs` vector would be destructed, along with the second `Descriptor` pointer it contains. The destructor of the `Node` (`l:pk(xpub...)`) it contains would in turn be called, moving out all its subs that are also pointed to by the first `Descriptor`'s `Node`. As soon as you try to perform an operation on it it will try to access moved structures.